streamwebs_frontend/streamwebs/views.py

```diff
@@ -244,3 +246,66 @@ def water_quality_edit(request, site_slug):
     return render(request, 'streamwebs/datasheets/water_quality_edit.html', {
         'site': site
     })
+
+
+def riparian_transect_view(request, site_slug, data_id):
+    transects = RiparianTransect.objects.filter(site_id=site_slug)
+    transect = transects.get(id=data_id)
+    zones = TransectZone.objects.filter(transect_id=transect)
+    site = Site.objects.get(id=site_slug)
+
+    # Invoking the database by evaluating the queryset before passing it to the
+    # template is necessary in order to pass Travis tests.
+    # https://docs.djangoproject.com/en/1.9/ref/models/querysets/#when-querysets-are-evaluated
+    zones = list(zones)
+
+    return render(
+        request, 'streamwebs/datasheets/riparian_transect_view.html', {
+            'transect': transect,
+            'zones': zones,
+            'site': site
+            }
+        )
+
+
+def riparian_transect_edit(request, site_slug):
+    """
+    The view for the submission of a new riparian transect data sheet.
+    """
+    added = False
+    site = Site.objects.get(id=site_slug)
```

Comment:
@lo9key @alxngyn Great question! Rather than Django magic, this is our bad-- we previously were undecided about whether we should use site slugs or the data sheet object ids in our urls. This line assumes that the site_slug that gets passed in is indeed a number, i.e. the data sheet id. @athai has decided just recently that we should start using ids; the fix will probably be in its own PR.